daemon/cmd/ipam.go

```diff
@@ -353,3 +365,41 @@ func (d *Daemon) bootstrapIPAM() {
 	d.ipam = ipam.NewIPAM(d.datapath.LocalNodeAddressing(), option.Config, d.nodeDiscovery, d.k8sWatcher)
 	bootstrapStats.ipam.End(true)
 }
+
+func (d *Daemon) parseHealthEndpointInfo(result *ipam.AllocationResult) error {
```

Comment:
Is there any chance of further sharing this implementation with `plugins/cilium-cni/eni.go:eniAdd()`?

Reply:
I wonder if we could try the interfaces approach from above, where the `ipam.AllocationResult` object implements a bunch of getters for stuff like the Gateway IP, CIDRs, MAC and then a function in `pkg/aws/eni/routing` like `NewRoutingInfo()` accepts an interface for any object that implements these functions, and uses these to construct the RoutingInfo object, which the daemon can then just set like `d.healthEndpointRouting = enirouting.NewRoutingInfo(result)`.

The cilium-cni code could then do a similar thing with `models.IPAMAddressResponse` (or probably more likely, a wrapper for this since the `models` code tends to be autogenerated), and furthermore you could also pass a dummy implementation into the enirouting package for testing purposes.